### PR TITLE
[refactor](fe-core src test catalog): refactor and replace use NIO

### PR DESCRIPTION
**Pull request by dinggege1024**
# Proposed changes

Issue Number: [#12780](https://github.com/apache/doris/issues/11768) 

## Problem summary

refactor and replace use NIO

## Checklist(Required)

1. Does it affect the original behavior: 
    - [✅] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [✅] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [✅] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [✅] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [✅] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Why changing this? Is there any explaination?

**Comment by dinggege1024** [comment]
> Why changing this? Is there any explaination?

Updated in proposed Changes

File: fe/fe-core/src/test/java/org/apache/doris/catalog/AdminStmtTest.java [hunk 1]
```diff
@@ -20,21 +20,19 @@
...
 import org.apache.doris.catalog.MaterializedIndex.IndexExtState;
 import org.apache.doris.catalog.Replica.ReplicaStatus;
-import org.apache.doris.common.AnalysisException;
 import org.apache.doris.common.Pair;
 import org.apache.doris.persist.SetReplicaStatusOperationLog;
 import org.apache.doris.utframe.TestWithFeService;
 
 import com.google.common.collect.Lists;
-import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
```
Comment (jackwener) [inline fe/fe-core/src/test/java/org/apache/doris/catalog/AdminStmtTest.java:28]: Good👍, Nereids already use junit 5 allover.

File: fe/fe-core/src/test/java/org/apache/doris/catalog/ColumnGsonSerializationTest.java [hunk 2]
```diff
@@ -117,6 +111,9 @@ public void testSerializeColumnList() throws IOException, AnalysisException {
         Assert.assertEquals(c1, columns.get(0));
         Assert.assertEquals(c2, columns.get(1));
         Assert.assertEquals(c3, columns.get(2));
+        // 3. Delete files
+        in.close();
```
Comment (jackwener) [inline fe/fe-core/src/test/java/org/apache/doris/catalog/ColumnGsonSerializationTest.java:115]: 👍

**Review by jackwener — DISMISSED** [review]
LGTM

**Comment by jackwener** [comment]
Thanks @dinggege1024

File: fe/fe-core/src/test/java/org/apache/doris/catalog/ColumnStatsTest.java [hunk 3]
```diff
@@ -59,29 +57,29 @@ public void testSerialization() throws Exception {
...
-        Assert.assertFalse(rStats3.equals(this));
-        Assert.assertFalse(rStats2.equals(rStats3));
+        Assert.assertEquals(rStats3, rStats3);
+        Assert.assertNotEquals(rStats3, this);
+        Assert.assertNotEquals(rStats2, rStats3);
 
         // 3. delete files
         dis.close();
-        file.delete();
+        Files.exists(path);
```
Comment (jackwener) [inline fe/fe-core/src/test/java/org/apache/doris/catalog/ColumnStatsTest.java:82]: deleteIfExists

**Comment by dinggege1024** [comment]
> Why changing this? Is there any explaination?

**Review by jackwener — APPROVED** [review]
LGTM

**Comment by jackwener** [comment]
Thanks @dinggege1024

**Pull request merged**